test/resolveGitDir.spec.js

```diff
@@ -1,18 +1,28 @@
 import path from 'path'
 import resolveGitDir from '../src/resolveGitDir'
 
+/**
+ * resolveGitDir runs execa, so the mock needs to be disabled for these tests
+ */
+jest.unmock('execa')
+
 describe('resolveGitDir', () => {
-  it('should resolve to current working dir when .git is in the same dir', () => {
+  it('should resolve to current working dir when .git is in the same dir', async () => {
     const expected = process.cwd()
-    expect(resolveGitDir()).toEqual(expected)
+    expect(await resolveGitDir()).toEqual(expected)
   })
 
-  it('should resolve to the parent dir when .git is in the parent dir', () => {
+  it('should resolve to the parent dir when .git is in the parent dir', async () => {
     const expected = path.dirname(__dirname)
     const processCwdBkp = process.cwd
     process.cwd = () => __dirname
     // path.resolve to strip trailing slash
-    expect(path.resolve(resolveGitDir())).toEqual(expected)
+    expect(path.resolve(await resolveGitDir())).toEqual(expected)
     process.cwd = processCwdBkp
   })
+
+  it('should return null when not in a git directory', async () => {
+    const gitDir = await resolveGitDir({ cwd: '/' }) // assume root is not a git directory
+    expect(gitDir).toEqual(null)
```

Comment:
Not sure about that one. If it’s not in git directory we should not proceed at all. To me it should be an error thrown so the whole thing terminates. Thoughts?

Reply:
I checked that this was the behaviour of `find-parent-dir`: https://github.com/thlorenz/find-parent-dir/blob/master/index.js#L25

One alternative would be to throw, and then rethrow that in `runAll` in a similar way as the invalid config error.

Reply:
Sounds good to me

Reply:
It was a bit trickier than expected to test this change, since `execa` was globally mocked. I was however able to make the change! Runnin `lint-staged` outside a git directory now exits very fast with:

```sh
❯ ./lint-staged/index.js -c lint-staged/.lintstagedrc.json
Current directory is not a git directory!
```

Reply:
To be more specific, I left this to return `null`, but the `runAll` will throw if `gitDir` is missing.